### PR TITLE
Make InstallRuntimeComponentToFinalDestination use NativeOutputPath for copying to runtime output

### PR DESCRIPTION
**Pull request by elinor-fung**
The `InstallRuntimeComponentToFinalDestination` target from infrastructure for NativeAOT-ed runtime components (#98565) copies the component and symbol files to the runtime output directory. Make the target use `NativeOutputPath` for the for the library and symbol path instead of the ones copied to the publish directory. They should be available after `LinkNative`.

This should unblock https://github.com/dotnet/installer/pull/19333

This was working in the runtime repo because the projects under src/native/managed are currently using the ILCompiler package corresponding to the SDK being used to build the repo. That version of the package doesn't have https://github.com/dotnet/runtime/pull/98342 - the native-library.targets were relying the symbols already being copied before `_CopyAotSymbols` was run.

I think the infrastructure should use the live ILCompiler if possible, otherwise the version in Version.Details.xml/Version.props instead. This PR does not try to do that - will look into it separately.

### CODE REVIEW

**Review by lambdageek — APPROVED** [review]

File: src/native/managed/native-library.targets [hunk 1]
```diff
@@ -69,7 +69,7 @@
 
     <Target Name="InstallRuntimeComponentToFinalDestination"
             AfterTargets="CopyNativeBinary"
-            DependsOnTargets="CopyNativeBinary;StripLibraryLikeCoreCLRBuild" Condition="'$(IsRuntimeComponent)' == 'true'">
+            DependsOnTargets="CopyNativeBinary;_CopyAotSymbols;StripLibraryLikeCoreCLRBuild" Condition="'$(IsRuntimeComponent)' == 'true'">
```
[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/native/managed/native-library.targets:72]: Can it be copied from `bin\[config]\[framework]\[rid]\native` (`$(NativeOutputPath)`) instead? That one should be available after `LinkNative` already, we don't even need to run Publish. Looks like this is making three copies of the symbols. We could get away with two.

**Review by akoeplinger — APPROVED** [review]
LGTM to unblock the codeflow but I still see double writes in the binlog with this.

**Comment by lambdageek** [comment]
> I still see double writes in the binlog with this

Isn't that expected? `strip` overwrites its input

**Comment by akoeplinger** [comment]
> Isn't that expected? strip overwrites its input

The double write is in the output, i.e. the .pdb, and this is Windows so we don't run `strip` :)

**Pull request merged**